server/src/main/java/io/crate/metadata/cluster/DropPartitionsClusterStateTaskExecutor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.metadata.cluster;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.MetadataDeleteIndexService;
+import org.elasticsearch.index.Index;
+
+import io.crate.execution.ddl.tables.DropPartitionsRequest;
+
+public class DropPartitionsClusterStateTaskExecutor extends DDLClusterStateTaskExecutor<DropPartitionsRequest> {
```

Comment:
To me, the only reason to use a dedicated executor would be testing. Like it can be tested standalone without using the related action. Otherwise, I'd inline it. But maybe personal preference.

Reply:
No personal preference either, but I did inline it, since it makes it more easy to have one class and avoid code navigation.